delta/delta_1.4.1_swmm_catchment_areas.sql

```diff
@@ -0,0 +1,231 @@
+DROP VIEW IF EXISTS qgep_swmm.vw_subcatchments CASCADE;
+CREATE OR REPLACE VIEW qgep_swmm.vw_subcatchments AS
+SELECT
+  replace(ca.obj_id, ' ', '_') as Name,
+  ('raingage@' || replace(ca.obj_id, ' ', '_'))::varchar as RainGage,
+  coalesce(fk_wastewater_networkelement_rw_current, replace(ca.obj_id, ' ', '_')) as Outlet,
+  CASE
+    when surface_area is null then st_area(perimeter_geometry)
+    when surface_area < 0.01 then st_area(perimeter_geometry)
+    else surface_area
+  END as Area,
+  discharge_coefficient_rw_current as percImperv, -- take from catchment_area instead of default value
+  CASE
+    WHEN wn.situation_geometry IS NOT NULL
+    THEN 	
+    (
+    st_maxdistance(wn.situation_geometry, ST_ExteriorRing(perimeter_geometry))
+    + st_distance(wn.situation_geometry, ST_ExteriorRing(perimeter_geometry))
+    )/2
+    ELSE
+    (
+    st_maxdistance(st_centroid(perimeter_geometry), ST_ExteriorRing(perimeter_geometry))
+    + st_distance(st_centroid(perimeter_geometry), ST_ExteriorRing(perimeter_geometry))
+    )/2
+  END as Width, -- Width of overland flow path estimation
+  0.5 as percSlope, -- default value
+  0 as CurbLen, -- default value
+  NULL::varchar as SnowPack, -- default value
+  ca.identifier || ', ' || ca.remark as description,
+  ca.obj_id as tag,
+  ST_Simplify(ST_CurveToLine(perimeter_geometry), 5, TRUE) as geom
```

Comment:
Do we need them to be topologically correct still? Then ST_Simplify is likely not the right tool to use.

Reply:
I dont think that the topology play a role in SWMM. It is more the area which is important.

Reply:
Why don't we just take the topology as is? Any reason to simplify it?

Reply:
I don't remember why I simplified the geometry. There is probably no reason except to have a more schematic map in SWMM GUI. I will remove the simplification